### PR TITLE
Uuid generator

### PR DESCRIPTION
**Pull request by terox**
Hello every body,

I added an UUID generator as a new generator.

Tests could be better, but I am not a PHPSpec expert.

I hope that it helps

### CODE REVIEW

**Comment by kieljohn** [comment]
Hi Terox, thanks for the contribution, I'm a concerned that the StringGeneratorBundle could become a tangle of dependencies if they're all under the `require` node composer.json. I'm thinking this would be probably be better to be put under the `suggests`node

https://getcomposer.org/doc/04-schema.md#suggest

Then the package can use a UUID generator if that package has been installed, but it will not be installed by default.

I appreciate that this is already an issue with the `ircmaxell/random-lib` bundle.

I'll put it to the team however.

**Comment by terox** [comment]
Hi! I understand perfectly what are you saying. I will review it in a few days and I will move the dependencie as a composer suggest. I will add also a checking to check if the library is present or not to thrown an exception (good? :) ).

I don't understand the last line (may be my english is not enough good): 
> I appreciate that this is already an issue with the ircmaxell/random-lib bundle.

I will commit changes as soon as possible :)

Thank you

**Comment by kieljohn** [comment]
Thanks Terox, that sounds perfect 👍 , we do appreciate your contribution and understanding. Look forward to seeing the commit.

The second point was to say that although we already require the `ircmaxell/random-lib` library which is an external dependency, however this allows us to create secure random strings 'by default' rather than relying on php `rand` or `str_shuffle` functions and then suggesting people install the 'secure' version of it

**Comment by terox** [comment]
Hi again,

First of all sorry the delay. The last past few months were crazy.

I have moved the ramsey/uuid library as a suggested packaged and updated the README with some information about this generator.

Please check if all is fine or need a little more work or refactor. I am listening the suggestions.

Note: I think that we should install the suggested package (ramsey/uuid) to pass the tests?

Thank you so much

File: spec/Vivait/StringGeneratorBundle/Generator/UuidGeneratorSpec.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+use PhpSpec\ObjectBehavior;
+
+class UuidGeneratorSpec extends ObjectBehavior
+{
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Vivait\StringGeneratorBundle\Generator\UuidGenerator');
+    }
+
+    function let()
```
Comment (Brunty) [inline spec/Vivait/StringGeneratorBundle/Generator/UuidGeneratorSpec.php]: Remove empty function.

File: README.md [hunk 2]
```diff
@@ -49,6 +49,20 @@ Generates a secure random byte string using the `Symfony\Component\Security\Core
 @Generate(generator="secure_bytes", options={"length"=8})
 ```
 
+### `UuidStringGenerator`
+***For use this generator you should require the package ```ramsey/uuid``` in your application.***
+
+For generate a UUID v4 (or v1):
```
Comment (Brunty) [inline README.md]: Change to: "For generating a UUID"

File: src/Vivait/StringGeneratorBundle/Generator/UuidGenerator.php [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+    /**
+     * @var string
+     */
+    private $namespace;
+
+    /**
+     * Constructor.
+     */
+    public function __construct()
```
Comment (Brunty) [inline src/Vivait/StringGeneratorBundle/Generator/UuidGenerator.php]: Remove empty constructor

**Comment by Brunty** [comment]
👍  nice work, @terox!

**Comment by terox** [comment]
Thank you ;)

I will wait the tag to use in our production projects :)

**Pull request merged**